iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -46,16 +45,120 @@ protected String getTypeDescription() {
         return "Shunt compensator";
     }
 
+    class ShuntCompensatorLinearModelAdderImpl implements ShuntCompensatorLinearModelAdder {
+
+        private double bPerSection = Double.NaN;
+
+        private double gPerSection = Double.NaN;
+
+        private int maximumSectionCount = -1;
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setbPerSection(double bPerSection) {
+            this.bPerSection = bPerSection;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setgPerSection(double gPerSection) {
+            this.gPerSection = gPerSection;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setMaximumSectionCount(int maximumSectionCount) {
+            this.maximumSectionCount = maximumSectionCount;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorAdder add() {
+            ValidationUtil.checkbPerSection(ShuntCompensatorAdderImpl.this, bPerSection);
+            ValidationUtil.checkMaximumSectionCount(ShuntCompensatorAdderImpl.this, maximumSectionCount);
+            model = new ShuntCompensatorLinearModelImpl(bPerSection, gPerSection, maximumSectionCount);
+            return ShuntCompensatorAdderImpl.this;
+        }
+    }
+
+    class ShuntCompensatorNonLinearModelAdderImpl implements ShuntCompensatorNonLinearModelAdder {
+
+        private final TreeMap<Integer, ShuntCompensatorNonLinearModelImpl.SectionImpl> sectionsMap = new TreeMap<>();
+        private boolean hasDefinedG = false;
+
+        class SectionAdderImpl implements SectionAdder {
+
+            private int sectionIndex = -1;
+
+            private double b = Double.NaN;
+
+            private double g = Double.NaN;
+
+            @Override
+            public SectionAdder setSectionIndex(int sectionIndex) {
+                this.sectionIndex = sectionIndex;
+                return this;
+            }
+
+            @Override
+            public SectionAdder setB(double b) {
+                this.b = b;
+                return this;
+            }
+
+            @Override
+            public SectionAdder setG(double g) {
+                this.g = g;
+                return this;
+            }
+
+            @Override
+            public ShuntCompensatorNonLinearModelAdder endSection() {
+                if (sectionIndex == 0) {
+                    throw new ValidationException(ShuntCompensatorAdderImpl.this, "section 0 (b = 0.0, g = 0.0) should not be written, it corresponds to disconnected state");
```

Comment:
I fall in this exception with NGESO network.

Reply:
I will check but I think I will add a fix on CGMES converter here.